clients/src/main/java/org/apache/kafka/clients/NetworkClientUtils.java

```diff
@@ -56,20 +56,19 @@ public static boolean awaitReady(KafkaClient client, Node node, Time time, long
         long startTime = time.milliseconds();
         long expiryTime = startTime + timeoutMs;
 
-        if (!isReady(client, node, startTime))
-            client.ready(node, startTime);
+        if (isReady(client, node, startTime) ||  client.ready(node, startTime))
```

Comment:
That's relying a little on implementation details though. I think the proposed change makes sense from the perspective of not being too coupled to the implementation.